src/main/java/ai/elimu/rest/v2/crowdsource/WordContributionRestController.java

```diff
@@ -0,0 +1,189 @@
+            List<LetterToAllophoneMappingGson> wordGsonLetterToAllophoneMappingsGson = wordGson.getLetterToAllophoneMappings();
+            List<LetterToAllophoneMapping> letterToAllophoneMappings = new ArrayList<>();
+            for (LetterToAllophoneMappingGson letterToAllophoneMappingGson : wordGsonLetterToAllophoneMappingsGson) {
+                LetterToAllophoneMapping letterToAllophoneMapping = letterToAllophoneMappingDao.read(letterToAllophoneMappingGson.getId());
+                letterToAllophoneMappings.add(letterToAllophoneMapping);
+            }
+            newWord.setLetterToAllophoneMappings(letterToAllophoneMappings);
+            wordDao.create(newWord);
+
+            //Convert Request Body(String) to JSON Object.
+            JSONObject requestJson = new JSONObject(requestBody);
+
+            WordContributionEvent wordContributionEvent = new WordContributionEvent();
+            wordContributionEvent.setContributor(contributor);
+            wordContributionEvent.setTime(Calendar.getInstance());
+            wordContributionEvent.setWord(newWord);
+            wordContributionEvent.setRevisionNumber(newWord.getRevisionNumber());
+            wordContributionEvent.setComment(requestJson.getString("contributionComment"));
+            wordContributionEvent.setTimeSpentMs(System.currentTimeMillis() - Long.valueOf(requestJson.getString("timeStart")));
+            wordContributionEventDao.create(wordContributionEvent);
+
+            response.setStatus(HttpStatus.CREATED.value());
+        } catch (Exception ex) {
+            logger.error(ex);
+
+            jsonObject.put("result", "error");
+            jsonObject.put("errorMessage", ex.getMessage());
+            response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
+        }
+
+        String jsonResponse = jsonObject.toString();
+        logger.info("jsonResponse: " + jsonResponse);
+        return jsonResponse;
+    }
+
+    /**
+     * This method will return all the required data to create a word. eg: Allophones etc.
+     * TODO : fetch other information that are optional when creating a new word eg: existing words(for Root word) etc.
+     *
+     * @param request
+     * @param response
+     * @return
+     */
+    @RequestMapping(value = "/word-data", method = RequestMethod.GET)
+    public String getWordDataForCrowdSourcing(
+            HttpServletRequest request,
+            HttpServletResponse response
+    ) {
+
+        JSONArray allophonesJsonArray = new JSONArray();
+        for (Allophone allophone : allophoneDao.readAllOrdered()) {
+            AllophoneGson allophoneGson = JpaToGsonConverter.getAllophoneGson(allophone);
+            String json = new Gson().toJson(allophoneGson);
+            allophonesJsonArray.put(new JSONObject(json));
+        }
+
+        String jsonResponse = allophonesJsonArray.toString();
+        logger.info("jsonResponse: " + jsonResponse);
+        return jsonResponse;
+    }
+
+}
```

Comment:
This entire method can probably be deleted if we create a separate REST controller for downloading a list of LetterToAllophoneMappings?
```suggestion
```

Reply:
Yes, I agree! Moving it to a separate controller would mean that we are adhering to proper REST priciples.